src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,572 @@
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetInvalidItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients/' . $timestamp);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+    }
+
+    public function testGetItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client' . $timestamp
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            'isActive' => true,
+            'maxParallelJobs' => 1,
+            'name' => 'client' . $timestamp,
+            'owner' => 1,
+            'postScripts' => [],
+            'preScripts' => [],
+            'quota' => '-1',
+            'rsyncLongArgs' => null,
+            'rsyncShortArgs' => null,
+            'sshArgs' => null,
+            'url' => ""
+        ]);
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'description' => 'description',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client'.$timestamp.'0',
+                'owner' => 1,
+                'postScripts' => [],
+                'preScripts' => [],
+                'quota' => - 1,
+                'rsyncLongArgs' => '',
+                'rsyncShortArgs' => '',
+                'sshArgs' => '',
+                'url' => 'root@172.17.0.1'
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, ['name' => 'client'.$timestamp.'0']);
+        
+        $httpClient->request('PUT', $iri, [
+            'json' => [
+                'description' => 'description updated',
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client',
```

Comment:
Why not just append something to the original timestamped name? Otherwise you might collide with an already existing client name.